FE won't quit when disk is full

<!-- (At least include the following, feel free to add if you have more content) -->

```
2022-08-18 09:07:05,290 ERROR (replayer|73) [GlobalStateMgr$5.runOneCycle():1576] replayer thread catch an exception when replay journal.
com.starrocks.journal.JournalException: failed to get DB names for 3 times!
        at com.starrocks.journal.bdbje.BDBJournalCursor.wrapDatabaseException(BDBJournalCursor.java:82) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.refresh(BDBJournalCursor.java:160) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1565) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1631) [starrocks-fe.jar:?]
Caused by: com.sleepycat.je.LogWriteException: (JE 7.3.7) Environment must be closed, caused by: com.sleepycat.je.LogWriteException: Environment invalid because of previous exception: (JE 7.3.7) 172.26.195.97_9912_1659974025228(2):/home/disk1/sr/starrocks-20220808/fe-5c73402a-3343-4f5e-945a-c268a8de496b/meta/bdb java.io.IOException: No space left on device LOG_WRITE: IOException on write, log is likely incomplete. Environment is invalid and must be closed. Problem seen replaying entry Commit/13 vlsn=2,983,669 isReplicated="1"  Originally thrown by HA thread: REPLICA 172.26.195.97_9912_1659974025228(2)
        at com.sleepycat.je.LogWriteException.wrapSelf(LogWriteException.java:78) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.dbi.EnvironmentImpl.checkIfInvalid(EnvironmentImpl.java:1766) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.dbi.EnvironmentImpl.checkOpen(EnvironmentImpl.java:1775) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.Environment.checkOpen(Environment.java:2473) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.Environment.getDatabaseNames(Environment.java:2245) ~[je-7.3.7.jar:7.3.7]
        at com.starrocks.journal.bdbje.BDBEnvironment.getDatabaseNamesWithPrefix(BDBEnvironment.java:501) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.refresh(BDBJournalCursor.java:156) ~[starrocks-fe.jar:?]
        ... 3 more
Caused by: com.sleepycat.je.LogWriteException: Environment invalid because of previous exception: (JE 7.3.7) 172.26.195.97_9912_1659974025228(2):/home/disk1/sr/starrocks-20220808/fe-5c73402a-3343-4f5e-945a-c268a8de496b/meta/bdb java.io.IOException: No space left on device LOG_WRITE: IOException on write, log is likely incomplete. Environment is invalid and must be closed. Problem seen replaying entry Commit/13 vlsn=2,983,669 isReplicated="1"  Originally thrown by HA thread: REPLICA 172.26.195.97_9912_1659974025228(2)
        at com.sleepycat.je.log.FileManager.writeLogBuffer(FileManager.java:1704) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogBufferPool.writeBufferToFile(LogBufferPool.java:472) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogBufferPool.writeDirty(LogBufferPool.java:419) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.flushInternal(LogManager.java:1270) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.flushBeforeSync(LogManager.java:1239) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.FSyncManager.flushAndSync(FSyncManager.java:340) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.log(LogManager.java:343) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.txn.Txn.logCommitEntry(Txn.java:1008) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.txn.Txn.commit(Txn.java:767) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.txn.ReplayTxn.commit(ReplayTxn.java:246) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replay.replayEntry(Replay.java:530) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replica$ReplayThread.run(Replica.java:1213) ~[je-7.3.7.jar:7.3.7]
Caused by: java.io.IOException: No space left on device
        at java.io.RandomAccessFile.writeBytes(Native Method) ~[?:1.8.0_322]
        at java.io.RandomAccessFile.write(RandomAccessFile.java:525) ~[?:1.8.0_322]
        at com.sleepycat.je.log.FileManager.writeToFile(FileManager.java:1777) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.FileManager.writeLogBuffer(FileManager.java:1691) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogBufferPool.writeBufferToFile(LogBufferPool.java:472) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogBufferPool.writeDirty(LogBufferPool.java:419) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.flushInternal(LogManager.java:1270) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.flushBeforeSync(LogManager.java:1239) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.FSyncManager.flushAndSync(FSyncManager.java:340) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.log.LogManager.log(LogManager.java:343) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.txn.Txn.logCommitEntry(Txn.java:1008) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.txn.Txn.commit(Txn.java:767) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.txn.ReplayTxn.commit(ReplayTxn.java:246) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replay.replayEntry(Replay.java:530) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replica$ReplayThread.run(Replica.java:1213) ~[je-7.3.7.jar:7.3.7]
```

--- a/fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJournalCursor.java
+++ b/fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJournalCursor.java
@@ -24,6 +24,7 @@ package com.starrocks.journal.bdbje;
 import com.sleepycat.bind.tuple.TupleBinding;
 import com.sleepycat.je.DatabaseEntry;
 import com.sleepycat.je.DatabaseException;
+import com.sleepycat.je.EnvironmentFailureException;
 import com.sleepycat.je.LockMode;
 import com.sleepycat.je.OperationStatus;
 import com.sleepycat.je.rep.InsufficientLogException;
@@ -61,7 +62,7 @@ public class BDBJournalCursor implements JournalCursor {
      *
      * 1. wrap as JournalException and return if it's a normal DatabaseException.
      * 2. RestartRequiredException is a fatal error since we're replaying as a follower, we must exit by raising an
-     * JournalInconsistentException.
+     * JournalInconsistentException. Same with EnvironmentFailureException when replicated environment is invalid.
      * 3. If it is an InsufficientLogException, we should call refreshLog() to restore the data as much as possible.
      */
     protected JournalException wrapDatabaseException(DatabaseException originException, String errMsg)
@@ -74,6 +75,13 @@ public class BDBJournalCursor implements JournalCursor {
                 // then exit the process because we may have read dirty data.
                 environment.refreshLog((InsufficientLogException) originException);
             }
+            JournalInconsistentException journalInconsistentException = new JournalInconsistentException(errMsg);
+            journalInconsistentException.initCause(originException);
+            throw journalInconsistentException;
+        } else if (originException instanceof EnvironmentFailureException
+                && ! environment.getReplicatedEnvironment().isValid()) {
+            errMsg += "Got EnvironmentFailureException and the current ReplicatedEnvironment is invalid, will exit.";
+            LOG.warn(errMsg, originException);
             JournalInconsistentException journalInconsistentException = new JournalInconsistentException(errMsg);
             journalInconsistentException.initCause(originException);
             throw journalInconsistentException;

